File: solutions/javascript/MinimumNumberofArrowstoBurstBalloons/MinimumNumberofArrowstoBurstBalloons.js
/**
 * @param {number[][]} points
 * @return {number}
 */
var findMinArrowShots = function(points) {
    var count = 0, arrow = null;
    points.sort((x, y) => x[1] == y[1] ? x[0] - y[0] : x[1] - y[1]);
    for (var i = 0; i < points.length; i++) {
        if (arrow != null && points[i][0] <= arrow) {
            continue;
        }
        arrow = points[i][1];
        count++;
    }
    return count;
};

File: solutions/java/MinimumNumberofArrowstoBurstBalloons/MinimumNumberofArrowstoBurstBalloons.java
class Solution {
    public int findMinArrowShots(int[][] points) {
        if (points.length == 0) {
            return 0;
        }
        Arrays.sort(points, new Comparator<int[]>() {
            @Override
            public int compare(int[] o1, int[] o2) {
                return o1[1] == o2[1] ? o1[0] - o2[0] : o1[1] - o2[1];
            }
        });
        int count = 1;
        int arrow = points[0][1];
        for (int i = 1; i < points.length; i++) {
            if (points[i][0] > arrow) {
                arrow = points[i][1];
                count++;
            }
        }
        return count;
    }
}